pages/Livesite/announcements.js

```diff
@@ -1,24 +1,178 @@
+    if (activeHackathon) {
+      return subscribeToLivesiteAnnouncements(activeHackathon, (data) => {
+        setAnnouncements(data);
+      });
+    }
+  }, [activeHackathon, setAnnouncements]);
+
+  const handleNew = () => {
+    setCurrentAnnouncement({ editor: user });
+    setActiveModal(NEW);
+  };
+
+  const handleDelete = (key) => {
+    // eslint-disable-next-line no-restricted-globals
+    if (confirm('Are you sure? This cannot be undone')) {
+      deleteAnnouncement(activeHackathon, key);
+    }
+  };
+
+  const handleEdit = (key) => {
+    setCurrentAnnouncement({ ...announcements[key], editor: user, id: key });
+    setActiveModal(EDIT);
+  };
+
+  const handleCloseModal = () => {
+    if (currentAnnouncement.content) {
+      // eslint-disable-next-line no-restricted-globals
+      if (confirm('Are you sure? You will lose all progress')) {
+        setActiveModal('');
+      }
+    } else {
+      setActiveModal('');
+    }
+  };
+
+  const handleSave = () => {
+    updateAnnouncement(activeHackathon, currentAnnouncement);
+    setActiveModal('');
+  };
+
+  const changeCurrentAnnouncement = (content) => {
+    setCurrentAnnouncement({ ...currentAnnouncement, content });
+  };
+
+  return (
+    <Page
+      currentPath="Livesite"
+      hackathons={hackathons}
+      navbarItems={LIVESITE_NAVBAR}
+    >
+      <AnnouncementsCard
+        isLoading={isLoading}
+        announcements={announcements}
+        handleNew={handleNew}
+        handleDelete={handleDelete}
+        handleEdit={handleEdit}
+      />
+
+      <Modal
+        isOpen={activeModal === EDIT}
+        handleClose={handleCloseModal}
+        handleSave={handleSave}
+        modalAction={EDIT}
+      >
+        <div>
+          <strong>Announcement Content</strong>
+          <TextBox
+            defaultValue={currentAnnouncement.content}
+            modalAction={EDIT}
+            onChange={(event) => changeCurrentAnnouncement(event.target.value)}
+          />
+          <br />
+          <br />
+          <strong>Preview:</strong>
+          <Markdown content={currentAnnouncement.content} />
+          <p>
+            {format(currentAnnouncement.timestamp)} @{' '}
+            {announcementDateFormat(currentAnnouncement.timestamp)}
+          </p>
+        </div>
+      </Modal>
+
+      <Modal
+        isOpen={activeModal === NEW}
+        handleClose={handleCloseModal}
+        handleSave={handleSave}
+        modalAction={NEW}
+      >
+        <div>
+          <strong>Announcement Content</strong>
+          <TextBox
+            defaultValue={currentAnnouncement.content}
+            modalAction={NEW}
+            onChange={(event) => changeCurrentAnnouncement(event.target.value)}
+          />
+          <br />
+          <br />
+          <strong>Preview:</strong>
+          <Markdown content={currentAnnouncement.content} />
+          <p>
+            {format(currentAnnouncement.timestamp)} @{' '}
+            {announcementDateFormat(Date.now())}
```

Comment:
why are we using `Date.now()` here instead of `currentAnnouncement.timestamp`

Reply:
Timestamp doesn't exist for new announcements. I should change the line before